pysoa/client/client.py

```diff
@@ -362,10 +362,10 @@ def _expand_objects(self, objs_to_expand):
                             # If everything is okay, replace the expansion object with the response value
                             if exp_action_response.body:
                                 value = exp_action_response.body[expansion_node.response_field]
-                                # Add the expansion value to the object and remove the
-                                # source field.
-                                del exp_object[expansion_node.source_field]
-                                exp_object[expansion_node.dest_field] = value
+                                # Add the expansion value to the object
+                                # Assume there is one item, and discard the id-key
+                                (dest_obj, ) = value.values()
```

Comment:
Use `values()[0]` here rather than pattern matching.

Reply:
```TypeError: 'dict_values' object does not support indexing
```

Reply:
Yeah, `value.values()[0]` is Python 2-specific. Perhaps `list(value.values())[0]`, but, at that point, is that really the best way to do it? 🤔

Reply:
As long as we are _sure_ that there is only ever one value in this dict, then I would do it exactly like Yan has it. Performance is considerably better: 

```
In [13]: %timeit (dest_obj, ) = d.values()
151 ns ± 2.57 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [15]: %timeit dest_obj = list(d.values())[0]
352 ns ± 10.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

However, if there's any chance at all that this dict has more than one value (which I would argue calls into question this code in general), then it needs to be `list(value.values())[0]`.

Reply:
It is an invariant that there is one and only one item returned in `value`, because the request to `get_xxxs_by_ids` dispatches with a one-element array, thus the returned response should be a one-element collection too. Otherwise, there is a data inconsistency and it should fail explicitly with `ValueError` during unpacking.